src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -66,6 +68,37 @@ public Response indexDatasetTerms(List<DatasetTerm> datasets) throws IOException
     return esClient.performRequest(bulkRequest);
   }
 
+  public boolean validateQuery(String query) throws IOException {
+    Request validateRequest = new Request(
+        HttpMethod.GET,
+        "/" + esConfig.getDatasetIndexName() + "/_validate/query");
+    validateRequest.setEntity(new NStringEntity(query, ContentType.APPLICATION_JSON));
+    Response validateResponse = esClient.performRequest(validateRequest);
+
+    if (validateResponse.getStatusLine().getStatusCode() != 200) {
+      throw new IOException("Invalid Elasticsearch query");
+    }
+
+    String entityContent = new String(validateResponse.getEntity().getContent().readAllBytes(),
+        StandardCharsets.UTF_8);
+    var json = GsonUtil.getInstance().fromJson(entityContent, Map.class);
+
+    return (boolean) json.get("valid");
+  }
+
+  public Response searchDatasets(String query) throws IOException {
+    if (!validateQuery(query)) {
+      throw new IOException("Invalid Elasticsearch query");
+    }
+
+    Request searchRequest = new Request(
+        HttpMethod.GET,
+        "/" + esConfig.getDatasetIndexName() + "/_search");
+    searchRequest.setEntity(new NStringEntity(query, ContentType.APPLICATION_JSON));
+
+    return esClient.performRequest(searchRequest);
```

Comment:
I'm seeing a `JerseyGsonProvider` exception parsing the response for what should be a valid query
```
{ "query": { "query_string": { "query": "(GRU) OR (HMB)" } } }
```
...
```
consent  | ERROR [2023-07-24 13:28:17,461] io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a request: 9db3ef52a2d8eada
consent  | ! java.lang.reflect.InaccessibleObjectException: Unable to make field final byte[] java.nio.ByteBuffer.hb accessible: module java.base does not "opens java.nio" to unnamed module @3b4ce44c
consent  | ! at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)
consent  | ! at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)
consent  | ! at java.base/java.lang.reflect.Field.checkCanSetAccessible(Unknown Source)
consent  | ! at java.base/java.lang.reflect.Field.setAccessible(Unknown Source)
consent  | ! at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:35)
consent  | ! ... 84 common frames omitted
```